File: 1523.count-odd-numbers-in-an-interval-range/solution.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        count = 0
        
        if high & 1 == 1:
            count += 1
            high -= 1
        
        if (low >= high):
            return count
        
        if low & 1 == 1:
            count += 1
            low += 1
        
        count += (high - low) >> 1
        return count

File: 1523.count-odd-numbers-in-an-interval-range/solution.kt
class Solution {
    fun countOdds(low: Int, high: Int): Int {
        if (low.rem(2) == 0 && high.rem(2) == 0) {
            return (high - low) / 2
        } else {
            return (high - low) / 2 + 1
        }
    }
}
